sova/dashboard/services/settings_service.py

```
@@ -47,35 +45,52 @@ def _flatten_dict(prefix: str, obj: dict, result: dict, registered: frozenset[st
+        doc = tomlkit.parse(toml_path.read_text())
+
+        parts = key.split(".")
+        target = doc
+        for part in parts[:-1]:
+            if part not in target:
+                target[part] = tomlkit.table()
+            target = target[part]
+
+        target[parts[-1]] = _cast_value(value)
+        toml_path.write_text(tomlkit.dumps(doc))
         return {"status": "ok", "key": key, "value": value}
```

Ruff (0.16.1)

[warning] Consider moving this statement to an `else` block

(TRY300)